### PR TITLE
[WIP] PIM-5156: Create a Buffer component and use it in csv products export

### PR DESCRIPTION
**Pull request by phaseinducer**
https://akeneo.atlassian.net/browse/PIM-5156

| Q                 | A
| ----------------- | ---
| Specs             |
| Behats            |
| Blue CI           |
| Changelog updated |
| Review and 2 GTM  |
| Micro Demo (PO)   |
| Migration script  |
| Tech Doc          |

This PR is an improvement of https://github.com/akeneo/pim-community-dev/pull/3559.

### CODE REVIEW

File: src/Akeneo/Component/Buffer/BufferFactory.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * Basic implementation of BufferFactoryInterface
+ *
+ * @author    Yohan Blain <yohan.blain@akeneo.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ */
+class BufferFactory implements BufferFactoryInterface
+{
+    /** @var string */
+    private $className;
```
Comment (nidup) [inline src/Akeneo/Component/Buffer/BufferFactory.php]: private?

Comment (phaseinducer) [inline src/Akeneo/Component/Buffer/BufferFactory.php]: Components could be a good place to start to "privatize", if everybody is ok with that.

**Comment by jjanvier** [comment]
real question, no trolling
is this buffer thing big enough to create a component? can it become bigger? 
do we have cases where we can use it outside of the connectors?

**Comment by phaseinducer** [comment]
Replaced by https://github.com/akeneo/pim-community-dev/pull/3614

**Pull request closed without merging**